File: frontend/src/interfaces/ecommerce.ts
import { User } from "./users";
import { Organization } from "./organizations";

export type Product = {
  id: string;
  name: string;
  price: number;
  description: string;
  organization: Organization;
  maxBuyableQuantity: number;
}

export type PaymentStatus = "INITIATED" | "RESERVED" | "CAPTURED" | "CANCELLED" | "REFUNDED" | "FAILED" | "REJECTED";

export type Order = {
  id: string;
  product: Product;
  user: User;
  quantity: number;
  totalPrice: number;
  paymentStatus: PaymentStatus;
  date: string;
  authToken: string;
  paymentAttempt: number;
}

File: backend/utils/testing/factories/ecommerce.py
import random
import factory
from apps.ecommerce.models import Order, Product
from factory.django import DjangoModelFactory
from faker import Factory

from utils.testing.factories.organizations import OrganizationFactory


fake = Factory.create()
cars = ["Luna", "Nova", "Atmos", "Eld", "Gnist", "Vilje", "Arctos", "Aquilo", "Borealis"]  # Atmos <3


class ProductFactory(DjangoModelFactory):
    class Meta:
        model = Product

    name = fake.first_name() + " " + random.choice(cars)
    price = 1000.00
    description = fake.sentence(6)
    organization = factory.SubFactory(OrganizationFactory)
    total_quantity = 5
    max_buyable_quantity = 2


class OrderFactory(DjangoModelFactory):
    class Meta:
        model = Order

    quantity = 5
    total_price = 5000.00
